.github/workflows/ci.yaml

```
@@ -203,10 +203,17 @@ jobs:
     # The job name should match the name of the `nox-cross-arch` job.
     name: Cross-arch tests with nox
     needs: ["nox-cross-arch"]
+    # We only run if there are failures, to propagate the failure, so if this
+    # check is required, it will fail if the child matrix jobs failed.
+    # If they didn't ran, this job will be skipped because there will be no
+    # dependency with a failure result.
```

I'd be explicit to refer to `the child matrix jobs` rather than using `they`. Also changed s/ran/run. If we agree on this, then we can apply it on similar changes

```suggestion
    # If the child matrix jobs didn't run, this job will be skipped
    # because there will be no dependency with a failure result.
```